src/Impostor.Server/Net/Inner/Objects/GameManager/InnerGameManager.cs

```diff
@@ -53,12 +53,20 @@ internal int GetGameLogicTag<T>(T logic)
 
     public override async ValueTask<bool> HandleRpcAsync(ClientPlayer sender, ClientPlayer? target, RpcCalls call, IMessageReader reader)
     {
+        // Let all logic components process this RPC. If at least one component handles it, return true.
+        var result = false;
         foreach (var logicComponent in _logicComponents)
         {
-            await logicComponent.HandleRpcAsync(call, reader);
+            result |= await logicComponent.HandleRpcAsync(call, reader);
```

Comment:
Nifty

Reply:
Yeah, I tried to keep the behavior of the previous code intact and `|=` doesn't short circuit iirc